controllers/quay/quayregistry_controller_test.go

```diff
@@ -171,31 +175,35 @@ var _ = Describe("Reconciling a QuayRegistry", func() {
 				To(Succeed())
 		})
 
-		It("should not generate a self-signed TLS cert/key pair in a new `Secret`", func() {
-			// Reconcile again to get past defaulting step
-			result, err = controller.Reconcile(context.Background(), reconcile.Request{NamespacedName: quayRegistryName})
+		// This test needs to be fixed. Since the ObjectStorage API is not available, we cannot use managed storage.
```

Comment:
Wasn't this test passing before your changes? What exactly happened that made this test to break?

Reply:
This PR is to block the rollout of further resources when the provided config.yaml is invalid for the provided spec. We do not want the Operator to continue reconcliing if the given config.yaml is missing fields for unmanaged components. The testing suite in the operator is essentially running against an etcd instance, so there is no ObjectStorage API available. This means that the current tests in this step run against a spec with unmanaged object storage, and so the generated config won't have a DISTRIBUTED_STORAGE_CONFIG. This missing field will block the rollout of the TLS secret.